### PR TITLE
`Development`: Improve selectors used in e2e cypress tests for exam assessment

### PR DESCRIPTION
**Pull request by rs-maker**
### Motivation and Context
 We are using selectors in Cypress tests that are not ids. These selectors might change during development and we therefore want to use more stable id selectors.
In this PR i fix all the bad selectors used in Exam Assessment specs and related page objects by adding ids where needed. I also remove all assertions asserting notificiations as that is not an important part of cypress testing.

### Steps for Testing

We have a tutorial for testing cypress PRs <a href="https://github.com/ls1intum/Artemis/blob/develop/src/test/cypress/README.md"> here </a>

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts [hunk 1]
```diff
@@ -84,12 +85,12 @@ describe('Exam assessment', () => {
 
         it('Assess a programming exercise submission (MANUAL)', () => {
             cy.login(tutor, '/course-management/' + course.id + '/exams');
-            cy.contains('Assessment Dashboard', { timeout: examDuration }).click();
+            examManagement.openAssessmentDashboard(exam.id, 120000);
```
Comment (jhainzlschmid) [inline src/test/cypress/integration/exam/ExamAssessment.spec.ts:88]: Is there a reason why you changed the timeout from `examDuration` to a static 2 minutes?

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts [hunk 2]
```diff
@@ -84,12 +85,12 @@ describe('Exam assessment', () => {
...
             startAssessing();
             examAssessment.addNewFeedback(2, 'Good job');
             examAssessment.submit();
             cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
-            cy.get('.question-options').contains('6.6 of 10 points').should('be.visible');
+            cy.get('#result-score').should('contain.text', '6.6 of 10 points');
```
Comment (jhainzlschmid) [inline src/test/cypress/integration/exam/ExamAssessment.spec.ts:93]: I think its still important to verify that the score is actually shown to the user. The element could contain the text, but not be visible.
```suggestion
            cy.get('#result-score').should('contain.text', '6.6 of 10 points').and('be.visible');
```

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts [hunk 3]
```diff
@@ -129,7 +130,7 @@ describe('Exam assessment', () => {
                     expect(assessmentResponse.response?.statusCode).to.equal(200);
```
Comment (jhainzlschmid) [inline src/test/cypress/integration/exam/ExamAssessment.spec.ts:130]: We should use strongly typed code.
```typescript
examAssessment.submitModelingAssessment().then((assessmentResponse: Interception) => {
    expect(assessmentResponse.response!.statusCode).to.equal(200);
});
```

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts [hunk 4]
```diff
@@ -129,7 +130,7 @@ describe('Exam assessment', () => {
...
                 });
                 cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
-                cy.contains('4 of 10 points').should('be.visible');
+                cy.get('#result-score').should('contain.text', '4 of 10 points');
```
Comment (jhainzlschmid) [inline src/test/cypress/integration/exam/ExamAssessment.spec.ts:133]: We should also still assert that the element is actually visible

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts [hunk 5]
```diff
@@ -152,14 +153,14 @@ describe('Exam assessment', () => {
...
-                cy.contains('Assessment Dashboard', { timeout: 60000 }).click();
+                examManagement.openAssessmentDashboard(exam.id, 60000);
                 startAssessing();
                 examAssessment.addNewFeedback(7, 'Good job');
                 examAssessment.submitTextAssessment().then((assessmentResponse) => {
                     expect(assessmentResponse.response?.statusCode).to.equal(200);
                 });
                 cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
-                cy.get('.question-options').contains('7 of 10 points').should('be.visible');
+                cy.get('#result-score').should('contain.text', '7 of 10 points');
```
Comment (jhainzlschmid) [inline src/test/cypress/integration/exam/ExamAssessment.spec.ts:163]: Visibility should also be asserted here

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts [hunk 6]
```diff
@@ -197,18 +198,17 @@ describe('Exam assessment', () => {
                 cy.wait(examEnd.diff(dayjs(), 'ms'));
             }
             cy.login(student, '/courses/' + course.id + '/exams/' + exam.id);
-            const score = '5 of 10 points';
             // Sometimes the feedback fails to load properly on the first load...
-            cy.reloadUntilFound(`jhi-result:contains(${score})`);
-            cy.get('jhi-result').contains(score).should('be.visible');
+            cy.reloadUntilFound(`#result-score`);
+            cy.get('#result-score').should('contain.text', '5 of 10 points');
```
Comment (jhainzlschmid) [inline src/test/cypress/integration/exam/ExamAssessment.spec.ts:203]: Only declare the selector once and assert the visibility
```suggestion
            const resultSelector = '#result-score';
            cy.reloadUntilFound(resultSelector);
            cy.get(resultSelector).should('contain.text', '5 of 10 points').and('be.visible');
```

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts [hunk 7]
```diff
@@ -197,18 +198,17 @@ describe('Exam assessment', () => {
...
         });
     });
 
     function startAssessing() {
         artemis.pageobjects.assessment.course.clickExerciseDashboardButton();
         exerciseAssessment.clickHaveReadInstructionsButton();
         exerciseAssessment.clickStartNewAssessment();
-        cy.contains('You have the lock for this assessment').should('be.visible');
+        cy.get('#assessmentLockedCurrentUser').should('contain.text', 'You have the lock for this assessment');
```
Comment (jhainzlschmid) [inline src/test/cypress/integration/exam/ExamAssessment.spec.ts:211]: I think here we shouldn't assert the text in the element, but instead just make sure that the element is shown.
```suggestion
        cy.get('#assessmentLockedCurrentUser').should('be.visible');
```

File: src/test/cypress/support/pageobjects/assessment/CourseAssessmentDashboardPage.ts [hunk 8]
```diff
@@ -12,7 +12,7 @@ export class CourseAssessmentDashboardPage {
     }
 
     showTheComplaint() {
-        cy.get('.btn-primary').should('contain.text', 'Show the complaint').click();
+        cy.get('#show-complaint').should('contain.text', 'Show the complaint').click();
```
Comment (jhainzlschmid) [inline src/test/cypress/support/pageobjects/assessment/CourseAssessmentDashboardPage.ts:15]: Verifying the text of the complaint button only makes the test more unstable in my opinion
```suggestion
        cy.get('#show-complaint').click();
```

**Review by jhainzlschmid — CHANGES_REQUESTED** [review]
Found some minor things, mostly related to visibility assertions, which can be improved

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts (hunk 1 above)
Comment (rs-maker) [reply src/test/cypress/integration/exam/ExamAssessment.spec.ts:88]: I just didnt remember this when i refactored all the clicks.

File: src/test/cypress/integration/exam/ExamAssessment.spec.ts (hunk 3 above)
Comment (rs-maker) [reply src/test/cypress/integration/exam/ExamAssessment.spec.ts:130]: 👍

File: src/test/cypress/support/commands.ts [hunk 9]
```diff
@@ -35,8 +35,8 @@ declare global {
             login(credentials: CypressCredentials, url?: string): any;
             logout(): any;
             loginWithGUI(credentials: CypressCredentials): any;
-            getSettled(selector: string, options?: {}): Chainable<Cypress>;
-            reloadUntilFound(selector: string): Chainable<Cypress>;
+            getSettled(selector: string, options?: {}): Chainable<unknown>;
+            reloadUntilFound(selector: string, interval?: number, timeout?: number): Chainable<undefined>;
```
Comment (Strohgelaender) [inline src/test/cypress/support/commands.ts:39]: You used once `unknown` and once `undefined` here, is this on purpose?

Comment (rs-maker) [reply src/test/cypress/support/commands.ts:39]: They are the types returned by the custom commands, if i change them to something more explicit like Chainable<Interception> the interpreter complains that they dont match.

**Review by jhainzlschmid — APPROVED** [review]
[Tests passed](https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETBBT-740) and my review suggestions have been implemented

**Pull request merged**